lib/gitsh/tab_completion/dsl.rb

```diff
@@ -0,0 +1,16 @@
+require 'gitsh/tab_completion/dsl/lexer'
+require 'gitsh/tab_completion/dsl/parser'
+
+module Gitsh
+  module TabCompletion
+    module DSL
+      def self.load(path, start_state, env)
+        source = File.read(path)
+        tokens = Lexer.lex(source)
+        factory = Parser.parse(tokens, gitsh_env: env)
+        factory.build(start_state)
+      rescue Errno::ENOENT
+      end
```

Comment:
I could move the rescue up a level to the `Gitsh::TabCompletion::AutomatonFactory` object?

I don't think I want a missing configuration file to cause an error that the user sees, but maybe it should—without the file there will be no tab completion.